attribute.go

```diff
@@ -24,6 +24,30 @@ type Attribute struct {
 	Data []byte
 }
 
+// #define NLA_F_NESTED
+const nlaNested uint16 = 0x8000
+// #define NLA_F_NET_BYTE_ORDER
+const nlaNetByteOrder uint16 = 0x4000
+// Masks all bits except for Nested and NetByteOrder
+const nlaTypeMask = ^(nlaNested | nlaNetByteOrder)
+
+// IsNested determines if an Attribute's data contains additional nested attributes.
+func (a Attribute) IsNested() bool {
+	return (a.Type & nlaNested) > 0
+}
+
+// IsNetByteOrder interprets the Type field of the Attribute to determine whether the attribute
+// is big endian (net byte order) or not.
+func (a Attribute) IsNetByteOrder() bool {
+	return (a.Type & nlaNetByteOrder) > 0
+}
+
+// GetType masks the Attribute's Type with the bits that are NOT
+// reserved for nlaNested and nlaNetByteOrder.
+func (a Attribute) GetType() uint16 {
```

Comment:
Regarding bloating the struct: I'm not too worried about it and think the usability improvement beats adding more methods to the API.